mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,709 @@
+                 data_prefix=None,
+                 test_mode=False,
+                 filename_tmpl='img_{:05d}.jpg',
+                 video_centric=True,
+                 reg_stats=None,
+                 body_segments=5,
+                 aug_segments=(2, 2),
+                 aug_ratio=(0.5, 0.5),
+                 clip_len=1,
+                 frame_interval=1,
+                 filter_gt=True,
+                 verbose=False):
+        super(SSNDataset, self).__init__(ann_file, pipeline, data_prefix,
+                                         test_mode)
+        self.train_cfg = train_cfg
+        self.test_cfg = test_cfg
+        self.assigner = train_cfg.ssn.assigner
+        self.sampler = train_cfg.ssn.sampler
+        self.verbose = verbose
+        self.filename_tmpl = filename_tmpl
+
+        if filter_gt or not test_mode:
+            valid_inds = self._filter_records()
+        print(f'{len(valid_inds)} out of {len(self.video_infos)}'
+              f'videos are valid.')
+        self.video_infos = [self.video_infos[i] for i in valid_inds]
+
+        # construct three pools:
+        # 1. Positive(Foreground)
+        # 2. Background
+        # 3. Incomplete
+        self.positive_pool = []
+        self.background_pool = []
+        self.incomplete_pool = []
+        self.construct_proposal_pools()
+
+        if reg_stats is None:
+            self.reg_stats = self._compute_regression_stats()
+        else:
+            self.reg_stats = reg_stats
+        self.video_centric = video_centric
+        self.body_segments = body_segments
+        self.aug_segments = aug_segments
+        self.aug_ratio = _pair(aug_ratio)
+        if not mmcv.is_tuple_of(self.aug_ratio, (int, float)):
+            raise TypeError(f'aug_ratio should be int, float'
+                            f'or tuple of int and float, '
+                            f'but got {type(aug_ratio)}')
+        assert len(self.aug_ratio) == 2
+
+        total_ratio = (
+            self.sampler.positive_ratio + self.sampler.background_ratio +
+            self.sampler.incomplete_ratio)
+        self.positive_per_video = int(
+            self.sampler.num_per_video *
+            (self.sampler.positive_ratio / total_ratio))
+        self.background_per_video = int(
+            self.sampler.num_per_video *
+            (self.sampler.background_ratio / total_ratio))
+        self.incomplete_per_video = (
+            self.sampler.num_per_video - self.positive_per_video -
+            self.background_per_video)
+
+        self.test_interval = self.test_cfg.ssn.sampler.test_interval
+        # number of consecutive frames
+        self.clip_len = clip_len
+        # number of steps (sparse sampling for efficiency of io)
+        self.frame_interval = frame_interval
+
+        # test mode or not
+        self.filter_gt = filter_gt
+        self.test_mode = test_mode
+
+        # yapf: disable
+        if self.verbose:
+            logger = get_root_logger()
```

Comment:
you can use set `self.logger` as a member variables to avoid calling `get_root_logger()` in many times.